beacon-chain/core/epoch/epoch_processing.go

```diff
@@ -156,25 +156,25 @@ func ProcessSlashings(state *stateTrie.BeaconState) (*stateTrie.BeaconState, err
 
 	// Compute the sum of state slashings
 	slashings := state.Slashings()
-	vals := state.Validators()
 	totalSlashing := uint64(0)
 	for _, slashing := range slashings {
 		totalSlashing += slashing
 	}
 
-	// Compute slashing for each validator.
-	for index, validator := range vals {
-		correctEpoch := (currentEpoch + exitLength/2) == validator.WithdrawableEpoch
-		if validator.Slashed && correctEpoch {
+	validatorFunc := func(idx int, val *ethpb.Validator) error {
```

Comment:
Can you add a comment on why call back is used here?

Reply:
can we instead just do
```go
if err := state.ApplyToEveryValidator(func(idx int, val *ethpb.Validator) error {
...
})
```